types/node.lua

```diff
@@ -271,50 +269,88 @@ function nodetypes.nnode:outputYourself (typesetter, line)
    end
 end
 
-function nodetypes.nnode:toText ()
+--- Returns the text content of the nnode.
+-- Contrary to the parent class, this is the actual text content of the node,
+-- not a text representation of the node.
+--
+-- @treturn string The text content of the nnode.
+function nnode:toText ()
    return self.text
 end
 
-nodetypes.unshaped = pl.class(nodetypes.nnode)
-nodetypes.unshaped.type = "unshaped"
+--- An unshaped node is a text node that has not been shaped yet.
+--
+-- Derived from `nnode`.
+--
+-- Properties is_unshaped is true.
+-- Note that is_nnode is NOT true, as an unshaped node is not a representable node yet.
+--
+-- @type unshaped
 
-function nodetypes.unshaped:_init (spec)
+local unshaped = pl.class(nnode)
+unshaped.type = "unshaped"
+
+--- Constructor
+--
+-- @tparam table spec A table with the properties of the unshaped node.
```

Comment:
Again those "undefined" `spec`.
The _only_ meaningful use is `SILE.types.node.unshaped({ text = "string", options = { ... }})`

We should make it clear what all these nodes take as contructor argument. Again I understand we can pass any key, and they'll end up in the instance via the generic box... But it's not clear what to expect, and it's certainly not a good object-oriented design, if _any_ node can be abused in so many ways.